geomesa-convert/geomesa-convert-avro/src/main/scala/org/locationtech/geomesa/convert/avro/AvroConverter.scala

```diff
@@ -75,15 +74,25 @@ object AvroConverter {
     * @return
     */
   def addBytes(schema: Schema): Schema = {
-    val fields = new java.util.ArrayList[Schema.Field](schema.getFields.size() + 1)
-    schema.getFields.asScala.foreach { field =>
-      fields.add(new Schema.Field(field.name, field.schema, field.doc, field.defaultValue()))
+    schema.getType match {
+      case Schema.Type.RECORD =>
+        val fields = new java.util.ArrayList[Schema.Field](schema.getFields.size() + 1)
+        schema.getFields.asScala.foreach { field =>
+          fields.add(new Schema.Field(field.name, field.schema, field.doc, field.defaultVal()))
+        }
+        fields.add(new Schema.Field(BytesField, Schema.create(Schema.Type.BYTES), "raw bytes", ""))
+
+        val updated = Schema.createRecord(schema.getName, schema.getDoc, schema.getNamespace, schema.isError)
+        updated.setFields(fields)
+        updated
+
+      case Schema.Type.UNION =>
+        Schema.createUnion(schema.getTypes.asScala.map(addBytes): _*)
```

Comment:
Wait!  I just saw this movie!  What happens if there are nested Unions!?!??!

Reply:
hopefully there aren't infinitely nested unions :P

Reply:
Well, it'd be due to bad bytes on disk...  Eh, we can just remember this for when the time comes...;)